bigquery/src/main/java/org/apache/zeppelin/bigquery/bigQueryInterpreter.java

```diff
@@ -0,0 +1,322 @@
+      credential = credential.createScoped(bigqueryScopes);
+    }
+
+    return new Bigquery.Builder(transport, jsonFactory, credential)
+        .setApplicationName("Zeppelin/1.0 (GPN:Apache Zeppelin;)").build();
+  }
+
+  //Function that generates and returns the schema and the rows as string
+  public static String printRows(final GetQueryResultsResponse response) {
+    StringBuilder msg = null;
+    msg = new StringBuilder();
+    for (TableFieldSchema schem: response.getSchema().getFields()) {
+      msg.append(schem.getName());
+      msg.append(TAB);
+    }      
+    msg.append(NEWLINE);
+    for (TableRow row : response.getRows()) {
+      for (TableCell field : row.getF()) {
+        msg.append(field.getV().toString());
+        msg.append(TAB);
+      }
+      msg.append(NEWLINE);
+    }
+    return msg.toString();
+  }
+
+  //Function to poll a job for completion. Future use
+  public static Job pollJob(final Bigquery.Jobs.Get request, final long interval) 
+      throws IOException, InterruptedException {
+    Job job = request.execute();
+    while (!job.getStatus().getState().equals("DONE")) {
+      System.out.println("Job is " 
+          + job.getStatus().getState() 
+          + " waiting " + interval + " milliseconds...");
+      Thread.sleep(interval);
+      job = request.execute();
+    }
+    return job;
+  }
+
+  //Function to page through the results of an arbitrary bigQuery request
+  public static <T extends GenericJson> Iterator<T> getPages(
+      final BigqueryRequest<T> requestTemplate) {
+    class PageIterator implements Iterator<T> {
+      private BigqueryRequest<T> request;
+      private boolean hasNext = true;
+      public PageIterator(final BigqueryRequest<T> requestTemplate) {
+        this.request = requestTemplate;
+      }
+      public boolean hasNext() {
+        return hasNext;
+      }
+      public T next() {
+        if (!hasNext) {
+          throw new NoSuchElementException();
+        }
+        try {
+          T response = request.execute();
+          if (response.containsKey("pageToken")) {
+            request = request.set("pageToken", response.get("pageToken"));
+          } 
+          else {
+            hasNext = false;
+          }
+          return response;
+        } 
+        catch (IOException e) {
+          e.printStackTrace();
+          return null;
+        }
+      }
+
+      public void remove() {
+        this.next();
+      }
+    }
+
+    return new PageIterator(requestTemplate);
+  }
+  
+  //Function to call bigQuery to run SQL and return results to the Interpreter for output
+  private InterpreterResult executeSql(String sql) {
+    int counter = 0;
+    StringBuilder finalmessage = null;
+    finalmessage = new StringBuilder("%table ");
+    String projId = getProperty(PROJECT_ID);
+    long wTime = Long.parseLong(getProperty(WAIT_TIME));
+    Iterator<GetQueryResultsResponse> pages = run(sql, projId, wTime);
+    while (pages.hasNext()) {
```

Comment:
Usually we want to handle errors here as well (I.e bad SQL syntax or network issues) and set paragraph error status to Error by returning `InterpreterResult(Code.ERROR)` in such cases.

I.e by looking at the code below, it seems that `pages` can be `null` in case of connectivity issues, resulting in the NPE instead.